tasks/image_size.js

```diff
@@ -7,53 +7,92 @@
 
 'use strict';
 
+var path = require('path');
+var chalk = require('chalk');
+var sizer = require('image-size');
+
 module.exports = function(grunt) {
-	var path = require('path');
-	var chalk = require('chalk');
-	var sizer = require('image-size');
-	var types = [
-		'.png',
-		'.gif',
-		'.bmp',
-		'.psd',
-		'.jpg',
-		'.tiff',
-		'.webp'
-	];
-
-	grunt.registerMultiTask('image_size', 'retrieve image saze information', function() {
-		var options = this.options({
-			configObject : ''
-		});
-
-		this.files.forEach(function (file) {
-			if ((!options.configObject || !options.configObject.length) && !file.dest) {
-				grunt.fail.warn('Neither options.configObject nor dest specified for this task.');
-				return;
-			}
-
-			var output = [];
-			file.src.forEach(function (src) {
-				if (grunt.file.exists(src) && grunt.file.isFile(src) && types.indexOf(path.extname(src).toLowerCase()) !== -1) {
-					var dimensions = sizer(src);
-					output.push({
-						name : src,
-						width : dimensions.width,
-						height : dimensions.height
-					});
-
-					grunt.verbose.writeln('dimensions of ' + src + ' width: ' + dimensions.width.toString() + ', height: ' + dimensions.height.toString());
-				}
-			});
-
-			if (options.configObject && options.configObject.length) {
-				grunt.config.set(options.configObject, output);
-			}
-
-			if (file.dest) {
-				grunt.file.write(file.dest, JSON.stringify(output));
-				grunt.log.writeln('File ' + chalk.cyan(file.dest) + ' created.');
-			}
-		});
-	});
+  grunt.registerMultiTask(
+    'image_size',
+    'Retrieve images size information',
+    function() {
+      var options = this.options({
+        processName: undefined,
+        processEntry: undefined,
+        processSizes: undefined,
+        configObject: undefined,
+        replacer: null,
+        space: 2
+      });
+
+      if (!this.files.length) {
+        return grunt.log.error('No files or `options.configObject` specified.');
+      }
+
+      var processedFiles = 0;
+
+      this.files.forEach(function(file) {
+        if (!file.src.length)
+          return grunt.log.error(
+            'No source files specified for ' + chalk.cyan(dest) + '.'
```

Comment:
dest -> file.dest and it can be undefined (if configObject is specified)

Reply:
Right, I've completely missed it. Fixed

Reply:
But it still can be undefined, need additional check for that (and another message, maybe)

Reply:
Am I missing something? I'm checking here is it defined and the message has been updated accordingly: https://i.imgur.com/NGdmCAD.png

Reply:
If file.dest is undefined (and file.src array is empty), then message in https://github.com/ArmorDarks/grunt-image-size/blob/feature/prettier/tasks/image_size.js#L34 will be like this: 'No source files specified for undefined.'

Reply:
Right.

I've added handling for this case by simply checking `file.dest` _before_ `file.src`.